src/main/java/com/pubnub/api/endpoints/objects_api/CompositeParameterEnricher.java

```diff
@@ -23,31 +46,47 @@ public static CompositeParameterEnricher createDefault() {
         final Limiter limiter = new Limiter();
         final TotalCounter totalCounter = new TotalCounter();
 
-        return new CompositeParameterEnricher(include, sorter, pager, filter, totalCounter, limiter);
+        return new CompositeParameterEnricher(include, sorter, pager, filter, totalCounter, limiter, null, null);
+    }
+
+    public static CompositeParameterEnricher createDefault(boolean includeStatusInParams, boolean includeTypeInParams) {
+        final Include include = new Include();
+        final Sorter sorter = new Sorter();
+        final Pager pager = new Pager();
+        final Filter filter = new Filter();
+        final Limiter limiter = new Limiter();
+        final TotalCounter totalCounter = new TotalCounter();
+        final IncludeParamValue includeStatus = new IncludeParamValue(STATUS, includeStatusInParams);
+        final IncludeParamValue includeType = new IncludeParamValue(TYPE, includeTypeInParams);
+
+        return new CompositeParameterEnricher(include, sorter, pager, filter, totalCounter, limiter, includeStatus, includeType);
```

Comment:
IMHO this would do what you want without creating additional classes:

```
 public static CompositeParameterEnricher createDefault() {
        return createDefault(false, false);
    }

    public static CompositeParameterEnricher createDefault(boolean includeStatusInParams, boolean includeTypeInParams) {
        final Include include = new Include();
        final Sorter sorter = new Sorter();
        final Pager pager = new Pager();
        final Filter filter = new Filter();
        final Limiter limiter = new Limiter();
        final TotalCounter totalCounter = new TotalCounter();
        if (includeStatusInParams) include.addInclusionFlag(STATUS);
        if (includeTypeInParams) include.addInclusionFlag(TYPE);

        return new CompositeParameterEnricher(include, sorter, pager, filter, totalCounter, limiter);
    }
```

Reply:
It's hard for me to admit but I this this option in better :)  I will modify code.

Reply:
Sorry that it took me so long to realise that there's easier way